From 9373a45580456383855896e9da6360ea08265aea Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Thu, 14 Mar 2019 03:55:54 -0700
Subject: [PATCH 0568/1852] v4l: xilinx-vpss-csc: Make max width and height
 mandatory properties

commit 1e89c06565af2e9e3f015a6edece2bec83146e2c from
https://github.com/Xilinx/linux-xlnx.git

This patch makes xlnx,max-width and xlnx,max-height as mandatory
properties. This breaks backward compatibility and a warning message is
added.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-vpss-csc.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-vpss-csc.c b/drivers/media/platform/xilinx/xilinx-vpss-csc.c
index fd0eedad53ee..0d8da2f7be27 100644
--- a/drivers/media/platform/xilinx/xilinx-vpss-csc.c
+++ b/drivers/media/platform/xilinx/xilinx-vpss-csc.c
@@ -957,7 +957,8 @@ static int xcsc_parse_of(struct xcsc_dev *xcsc)
 
 	rval = of_property_read_u32(node, "xlnx,max-height", &xcsc->max_height);
 	if (rval < 0) {
-		xcsc->max_height = XV_CSC_MAX_HEIGHT;
+		dev_err(dev, "xlnx,max-height is missing!");
+		return -EINVAL;
 	} else if (xcsc->max_height > XV_CSC_MAX_HEIGHT ||
 		   xcsc->max_height < XV_CSC_MIN_HEIGHT) {
 		dev_err(dev, "Invalid height in dt");
@@ -966,7 +967,8 @@ static int xcsc_parse_of(struct xcsc_dev *xcsc)
 
 	rval = of_property_read_u32(node, "xlnx,max-width", &xcsc->max_width);
 	if (rval < 0) {
-		xcsc->max_width = XV_CSC_MAX_WIDTH;
+		dev_err(dev, "xlnx,max-width is missing!");
+		return -EINVAL;
 	} else if (xcsc->max_width > XV_CSC_MAX_WIDTH ||
 		   xcsc->max_width < XV_CSC_MIN_WIDTH) {
 		dev_err(dev, "Invalid width in dt");
-- 
2.31.1

